File: Resources/Archive-2016/demos-news/SQLiteDemos/SQLiteDemos/Models/Superhero.m
//
//  Superhero.m
//  SQLiteDemos
//
//  Created by Doncho Minkov on 2/3/16.
//  Copyright © 2016 Doncho Minkov. All rights reserved.
//

#import "Superhero.h"

@implementation Superhero

-(id)initWithName:(NSString *)name {
    if(self = [super init]) {
        self.name = name;
    }
    return self;
}

+(Superhero *)superheroWithName:(NSString *)name {
    return [[Superhero alloc] initWithName:name];
}

@end

File: demos/SuperheroesApp/SuperheroesApp/SuperheroesApp/Models/Superhero.swift
//
//  Superhero.swift
//  SuperheroesApp
//
//  Created by Doncho Minkov on 3/31/17.
//  Copyright © 2017 Doncho Minkov. All rights reserved.
//

import UIKit

class Superhero {
    var id: String
    var name: String
    var secretIdentity: String
    
    var image: UIImage?
    
    var imgUrl: String
    
    convenience init(withName name: String,
         secretIdentity: String,
         andImageUrl imgUrl: String) {
        self.init(withId: "sh-\(arc4random())",
            name: name,
            secretIdentity: secretIdentity,
            andImageUrl: imgUrl)
    }
    init(withId id: String,
         name: String,
         secretIdentity: String,
         andImageUrl imgUrl: String) {
        self.id = id
        self.name = name
        self.secretIdentity = secretIdentity
        self.imgUrl = imgUrl
    }

}
